spec/spec_helper.rb

```diff
@@ -31,6 +31,22 @@ def travis?
   # https://relishapp.com/rspec/rspec-core/docs/command-line/only-failures
   config.example_status_persistence_file_path = 'tmp/example_status.txt'
 
+  config.before do |scenario|
+    selenium_ver = Selenium::WebDriver::VERSION
+
+    s_sym =  case selenium_ver.split('.')[0]
+             when "3" then :selenium3
+             when "4" then :selenium4
+             else
+               raise "Selenium #{selenium_ver} version is not supported"
+             end
+
+    metadata = scenario.metadata
+    if metadata[:selenium3] || metadata[:selenium4]
+      skip unless metadata[s_sym]
+    end
```

Comment:
Umm... you want to run/skip examples on ... like this?

| metadata[:selenium3] | metadata[:selenium4] | s_sym      | Run or Skip |
|:--                   |:--                   |:--         |:--          |
| false                | false                | :selenium3 | Run         |
| false                | true                 | :selenium3 | Skip        |
| true                 | false                | :selenium3 | Run         |
| true                 | true                 | :selenium3 | Run         |
| false                | false                | :selenium4 | Run       |
| false                | true                 | :selenium4 | Run         |
| true                 | false                | :selenium4 | Skip        |
| true                 | true                 | :selenium4 | Run         |

Reply:
yap!

Reply:
If so, then you can rewrite like this:

```
skip if (!metadata[:selenium3] && metadata[:selenium4] && s_sym == :selenium3) ||
        (metadata[:selenium3] && !metadata[:selenium4] && s_sym == :selenium4)
```

Reply:
OK, it smells but your code works as expected.

Reply:
Fine thanks.